pkg/oci/utils.go

```diff
@@ -353,42 +353,38 @@ func (spec *CompatOCISpec) PodID() (string, error) {
 func vmConfig(ocispec CompatOCISpec, config RuntimeConfig) (vc.Resources, error) {
 	resources := config.VMConfig
 
-	if ocispec.Linux == nil ||
-		ocispec.Linux.Resources == nil ||
-		ocispec.Linux.Resources.Memory == nil ||
-		ocispec.Linux.Resources.Memory.Limit == nil {
+	if ocispec.Linux == nil || ocispec.Linux.Resources == nil {
 		return resources, nil
 	}
 
-	memBytes := *ocispec.Linux.Resources.Memory.Limit
-
-	if memBytes <= 0 {
-		return vc.Resources{}, fmt.Errorf("Invalid OCI memory limit %d", memBytes)
+	if ocispec.Linux.Resources.Memory != nil &&
+		ocispec.Linux.Resources.Memory.Limit != nil {
+		memBytes := *ocispec.Linux.Resources.Memory.Limit
+		if memBytes <= 0 {
+			return vc.Resources{}, fmt.Errorf("Invalid OCI memory limit %d", memBytes)
+		}
+		// round up memory to 1MB
+		resources.Memory = uint((memBytes + (1024*1024 - 1)) / (1024 * 1024))
 	}
 
-	// round up memory to 1MB
-	resources.Memory = uint((memBytes + (1024*1024 - 1)) / (1024 * 1024))
-
-	if ocispec.Linux.Resources.CPU == nil ||
-		ocispec.Linux.Resources.CPU.Quota == nil ||
-		ocispec.Linux.Resources.CPU.Period == nil {
-		return resources, nil
-	}
+	if ocispec.Linux.Resources.CPU != nil &&
+		ocispec.Linux.Resources.CPU.Quota != nil &&
+		ocispec.Linux.Resources.CPU.Period != nil {
+		quota := *ocispec.Linux.Resources.CPU.Quota
+		period := *ocispec.Linux.Resources.CPU.Period
 
-	quota := *ocispec.Linux.Resources.CPU.Quota
-	period := *ocispec.Linux.Resources.CPU.Period
+		if quota <= 0 {
+			return vc.Resources{}, fmt.Errorf("Invalid OCI cpu quota %d", quota)
+		}
 
-	if quota <= 0 {
-		return vc.Resources{}, fmt.Errorf("Invalid OCI cpu quota %d", quota)
-	}
+		if period == 0 {
+			return vc.Resources{}, fmt.Errorf("Invalid OCI cpu period %d", period)
+		}
 
-	if period == 0 {
-		return vc.Resources{}, fmt.Errorf("Invalid OCI cpu period %d", period)
+		// round up to 1 CPU
```

Comment:
Just noting - we discussed in a call that we should ensure we understood what this was doing iirc (in particular when it was written in the original/expanded form with real values - like `10000 + (20000 - 1) / 20000`. I had a stare and think, and yes, this does the expected and works out how many 'whole cpus' the given quota gets us given the period - but the above `round up to 1 CPU` comment is slightly misleading - it is really rounding up to ensure we have `at least 1 CPU` aiui. Just for the record.

Reply:
Thanks for the analysis @grahamwhaley. I think you should submit a PR to fix the comment and maybe be more verbose about the reason why this is the right way to compute the number of CPUs.

Reply:
Actually that comment was add in #344 
/cc @wcwxyz